app/jobs/cache/record_counts/data_providers_job.rb

```diff
@@ -7,9 +7,24 @@ class DataProvidersJob < ActiveJob::Base
 
       queue_as :default
 
-      def perform(provider)
-        params = { query: '*:*', rows: 0, profile: 'minimal facets', facet: 'DATA_PROVIDER', qf: "PROVIDER:\"#{provider}\"" }
-        response = repository.search(params)
+      def perform(provider, collection_id = nil)
+        builder = search_builder(search_params_logic)
+        api_query = builder.rows(0).merge(query: '*:*', profile: 'minimal facets', facet: 'DATA_PROVIDER', qf: "PROVIDER:\"#{provider}\"")
+
+        cache_key = "browse/sources/providers"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.